src/main/java/io/fabric8/elasticsearch/plugin/OpenShiftElasticSearchService.java

```diff
@@ -23,82 +23,93 @@
 public class OpenShiftElasticSearchService extends AbstractLifecycleComponent
-        implements ConfigurationSettings {
+        implements ConfigurationSettings, LocalNodeMasterListener {
 
-    private final Logger logger;
-    private final UserProjectCache cache;
-    private final Settings settings;
+    private static final Logger LOGGER = Loggers.getLogger(OpenShiftElasticSearchService.class);
+    
+    private final PluginSettings settings;
+    private final ACLDocumentManager aclManager;
     private ScheduledThreadPoolExecutor scheduler;
 
     @SuppressWarnings("rawtypes")
     private ScheduledFuture scheduledFuture;
 
-    // @TODO - Determine if we should be using the Threadpool provided by Elastic
-    public OpenShiftElasticSearchService(final Settings settings, final Client esClient, final UserProjectCache cache) {
-        super(settings);
-        this.settings = settings;
-        this.logger = Loggers.getLogger(getClass(), settings);
-        this.cache = cache;
+    public OpenShiftElasticSearchService( 
+            final ACLDocumentManager aclManager, 
+            final PluginSettings pluginSettings) {
+        super(pluginSettings.getSettings());
+        this.settings = pluginSettings;
+        this.aclManager = aclManager;
     }
-
+    
     @Override
     protected void doStart() throws ElasticsearchException {
+        LOGGER.debug("Started");
+    }
 
-        boolean dynamicEnabled = settings.getAsBoolean(OPENSHIFT_DYNAMIC_ENABLED_FLAG,
-                OPENSHIFT_DYNAMIC_ENABLED_DEFAULT);
-        logger.debug("Starting with Dynamic ACL feature enabled: {}", dynamicEnabled);
-
+    @Override
+    public void onMaster() {
+        boolean dynamicEnabled = settings.isEnabled();
+        LOGGER.debug("Starting with Dynamic ACL feature enabled: {}", dynamicEnabled);
         if (dynamicEnabled) {
             // expiration thread
-            logger.debug("Starting the expiration thread...");
+            LOGGER.info("Starting the ACL expiration thread...");
             this.scheduler = (ScheduledThreadPoolExecutor) Executors.newScheduledThreadPool(1,
-                    EsExecutors.daemonThreadFactory(settings, "openshift_elasticsearch_service"));
+                    EsExecutors.daemonThreadFactory(settings.getSettings(), "openshift_elasticsearch_service"));
             this.scheduler.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
             this.scheduler.setContinueExistingPeriodicTasksAfterShutdownPolicy(false);
 
             Runnable expire = new Runnable() {
                 @Override
                 public void run() {
-                    cache.expire();
+                    aclManager.expire();
                 }
             };
-            this.scheduledFuture = this.scheduler.scheduleWithFixedDelay(expire, 5, 60, TimeUnit.SECONDS);
+            this.scheduledFuture = this.scheduler.scheduleWithFixedDelay(expire, 1000 * 60, settings.getACLExpiresInMillis(), TimeUnit.MILLISECONDS);
```

Comment:
what does this do?

Reply:
This schedules a repeating task to expire ACLs based on the config value.  The process begins 1 minute after loading.